GillianCore/engine/GeneralSemantics/General/GInterpreter.ml

```diff
@@ -587,7 +581,11 @@ struct
                | Some state' -> [ state' ]
                | _ -> [])
              fos)
-    | SpecVar xs -> [ State.add_spec_vars state (Var.Set.of_list xs) ]
+    | FreshSVar x ->
+        let new_svar = Generators.fresh_svar () in
+        let state' = State.add_spec_vars state (SS.singleton x) in
+        let v = State.eval_expr state' (LVar new_svar) in
+        [ update_store state' x v ]
```

Comment:
Should we check that `x` actually has the format of a program variable name?

Reply:
Probably yes. Ideally, we wouldn't use strings for variables and it would be type-sound...

Reply:
It's not done in Assignment, so I'm just not going to do it there.